### DISCUSSION TITLE
How to set the infinitesimal time interval in the evolve function?

### DISCUSSION THREAD

**Discussion by archisman-panigrahi** [QuSpin/QuSpin#494] [Q&A] [answered]
I am using the evolve function to time-evolve a wavefunction.
```
psi = H.evolve(psi_ 0 , 0 . 0 ,t_f)
```
I presume that the computer calculates the wavefunction at time t in the following way.
```
psi(t + delta_t) =  psi(t) - i H psi(t) delta_t
```
where `delta_t` is a very small number, and the computer iterates this process to calculate psi(t_f).

What is the default value of delta_t? Is there any way to set that?

**Comment by JiahaoYao** [comment]
plz check this [page](http://weinbe58.github.io/QuSpin/generated/quspin.tools.evolution.evolve.html)

**Comment by archisman-panigrahi** [comment]
@JiahaoYao I did see it, but could not figure out the answer to my question about the minimum time interval.

**Comment by mgbukov** [comment, marked as answer]
the `evolve` function uses scipy's Runge Kutta solver with an adaptive step size by default (`dop583`). You can change the integrator using the `solver_name` optional argument -- see the documentation.
